cupy/cuda/compiler.py

```diff
@@ -111,6 +112,14 @@ def compile_with_cache(source, options=(), arch=None, cache_dir=None,
     if arch is None:
         arch = _get_arch()
 
+    cuda_path = os.getenv('CUDA_PATH', None)
```

Comment:
Cloud you move this section to before [this line](https://github.com/cupy/cupy/blob/master/cupy/core/carray.pxi#L99)?